### DISCUSSION TITLE
Using Hichi IR TTL sensor via UART directly as Smart Meter for Dynamic Power Limiter

### DISCUSSION THREAD

**Discussion by PalmBSH** [hoylabs/OpenDTU-OnBattery#1153] [Q&A]
Is it possible to connect via UART a IR Hichi TTL sensor with openDTU oB? The protocol of the smart meter is SML. If it is possible, how do I connect it?

**Comment by ghost** [comment]
Hi PalmBSH,
yes it is possible to do so - I'm using this configuration and it's working fine!
You have to connect the RX-pin of your IR-TTL-sensor to your ESP on its pin 35 and use 3,3V for the power supply of the IR-sensor because the ESP doesn't accept any higher voltage on its input as signal.
Then you have to modify the the device profile within its pin_mapping.json file for defining the RX-pin and activating the right combination:

e.g.
[
    {
        "name": "Generic NodeMCU 32",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "powermeter": {
            "rx": 35,
            "tx": -1,
            "dere": -1
        }
    },
    {
        "name": "Generic NodeMCU 32 with SSD1306",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "display": {
            "type": 2,
            "data": 21,
            "clk": 22
        },
        "powermeter": {
            "rx": 35,
            "tx": -1,
            "dere": -1
        }
    },
    {
        "name": "Generic NodeMCU 32 with SH1106",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "display": {
            "type": 3,
            "data": 21,
            "clk": 22
        },
        "powermeter": {
            "rx": 35,
            "tx": -1,
            "dere": -1
        }
    }
]


That's it - have fun and enjoy OpenDTU-onBattery!

cheers
solarbastler

**Comment by spcqike** [comment]
as @solarbastler072024 said, yes it should work.

please have a look at the wiki under powermeter - SML
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter#sdm--serial-sml
